zerver/actions/message_send.py

```diff
@@ -690,11 +690,6 @@ def create_user_messages(
     return user_messages
 
 
-class ActivePresenceIdleUserData(TypedDict):
-    alerted: bool
-    notifications_data: UserMessageNotificationsData
```

Comment:
What's the motivation for removing the alert words logic here? We do want to implement notifications logic for them eventually, and it seems like removing this plumbing would make that future work more difficult; and I'm not sure what the trade-off is here.

Reply:
Whenever we do implement push/email notifications logic for alert words, I think we would anyways be adding new fields like `notifications_data.alert_word_email_notify` similar to the existing `notifications_data.wildcard_mention_email_notify`.

Once we do so, that makes this ad-hoc `ActivePresenceIdleUserData` structure unnecessary, because with the shortened code after this commit, we can the following unchanged:
```py
if user_notifications_data.is_notifiable(sender_id, idle=True):
    user_ids.add(user_notifications_data.user_id)
```

... and that would work because the `is_notifiable` method would internally check for alert words similar to how it checks for mentions.

Reply:
I added this explanation in the commit message.

Reply:
Cool, this makes sense.